### PR TITLE
Gossip JSON streams

### PR DESCRIPTION
**Pull request by tbekas**
* Gossip round is unidirectional
* Rumors are streamed as chunks
* Sending rumors as JSON

### CODE REVIEW

File: modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/GossipRoutes.scala [hunk 1]
```diff
@@ -1,53 +1,52 @@
...
+    case req @ POST -> Root / "inquiry" =>
       for {
-        endRequest <- req.as[EndGossipRoundRequest]
-        _ <- rumorQueue.offer(endRequest.answer)
-        _ <- metrics.updateRumorsReceived(endRequest.answer)
-        answer <- rumorStorage.getRumors(endRequest.inquiry)
-        endResponse = EndGossipRoundResponse(answer)
-        result <- Ok(endResponse)
-        _ <- metrics.updateRumorsSent(endResponse.answer)
+        endRequest <- req.as[RumorInquiryRequest]
```
Comment (marcinwadon) [inline modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/GossipRoutes.scala:37]: inquiryRequest

**Review by marcinwadon — DISMISSED** [review]

File: modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/GossipRoutes.scala (hunk 1 above)
Comment (tbekas) [reply modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/GossipRoutes.scala:37]: Fixing it.

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/PeerDiscovery.scala [hunk 2]
```diff
@@ -3,12 +3,15 @@ package org.tessellation.sdk.domain.cluster.programs
 import cats.effect.{Async, Ref}
```
Comment (tbekas) [inline modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/PeerDiscovery.scala:3]: All changes in this file will be discarded. I added these logs for the debug purposes only and they slipped in to the final commit.

File: modules/dag-shared/src/main/scala/org/tessellation/dag/snapshot/GlobalSnapshot.scala [hunk 3]
```diff
@@ -51,12 +50,6 @@ case class GlobalSnapshot(
 
 object GlobalSnapshot {
 
-  implicit def encoder[G[_]: KryoSerializer]: EntityEncoder[G, GlobalSnapshot] =
-    BinaryCodec.encoder[G, GlobalSnapshot]
-
-  implicit def decoder[G[_]: Concurrent: KryoSerializer]: EntityDecoder[G, GlobalSnapshot] =
-    BinaryCodec.decoder[G, GlobalSnapshot]
-
```
Comment (TheMMaciek) [inline modules/dag-shared/src/main/scala/org/tessellation/dag/snapshot/GlobalSnapshot.scala:59]: Don't we need binary codecs for returning GlobalSnapshot from L0 public endpoints? 🤔

Comment (tbekas) [reply modules/dag-shared/src/main/scala/org/tessellation/dag/snapshot/GlobalSnapshot.scala:59]: We need, but we don't need a special `EntityDecoder` instance for it. It's enough to import this in the server
```
import org.tessellation.ext.codecs.BinaryCodec.encoder
```
and this in the client
```
import org.tessellation.ext.codecs.BinaryCodec.decoder
```

File: modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/GossipDaemon.scala [hunk 4]
```diff
@@ -132,37 +115,35 @@ object GossipDaemon {
         private def runGossipRounds: F[Unit] =
           Stream
             .fromQueueUnterminated(selectedPeerQueue)
-            .parEvalMapUnordered(cfg.maxConcurrentRounds) { peer =>
-              runTimedGossipRound(peer).handleErrorWith { err =>
-                logger.error(err)(s"Error running gossip round with peer ${peer.show}") >>
-                  Spawn[F].start(healthcheck.triggerCheckForPeer(peer)).void
```
Comment (TheMMaciek) [inline modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/GossipDaemon.scala:138]: I guess we want to leave that bit ;)

**Review by kpudlik — APPROVED** [review]

**Review by TheMMaciek — APPROVED** [review]

File: modules/shared/src/main/scala/org/tessellation/shared/package.scala [hunk 5]
```diff
@@ -35,11 +35,9 @@ package object shared {
     classOf[NonEmptyList[_]] -> 305,
     classOf[Signed[_]] -> 306,
     classOf[AddressCache] -> 307,
-    classOf[PeerRumorBinary] -> 308,
-    classOf[StartGossipRoundRequest] -> 309,
-    classOf[StartGossipRoundResponse] -> 310,
-    classOf[EndGossipRoundRequest] -> 311,
-    classOf[EndGossipRoundResponse] -> 312,
+    classOf[PeerRumorRaw] -> 308,
```
[the remark this answers is not in the document]
Comment (tbekas) [reply modules/shared/src/main/scala/org/tessellation/shared/package.scala:38]: It's not a substitution, it's a rename. We no longer need the 309 and 310.

[the remark this answers is not in the document]
Comment (TheMMaciek) [reply modules/shared/src/main/scala/org/tessellation/shared/package.scala:38]: I guess the fields of these classes changed a bit as well 🤔 . I'm just trying to make sure that we won't run into troubles. Given that these are not a part of global snapshot, then I guess everything should work.

**Review by marcinwadon — APPROVED** [review]

File: modules/shared/src/main/scala/org/tessellation/shared/package.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (tbekas) [reply modules/shared/src/main/scala/org/tessellation/shared/package.scala:38]: Yep, it's not stuff serialized on disk, so it should be alright.

**Pull request merged**